### ISSUE TITLE
Does struct field renaming support private fields?

### ISSUE THREAD

**Issue by disconsented** [rustwasm/wasm-bindgen#2397]
I am trying to map across the [stripe.CreatePaymentMethod](https://stripe.com/docs/js/payment_methods/create_payment_method) function, but I am running into a bit of a blocker.

Because `type` is a keyboard it cannot be used as an attribute name thankfully #2360 was pulled in and I've ended up with the following:
```rust
#[wasm_bindgen]
pub struct PaymentMethodData{
    #[wasm_bindgen(js_name = type)]
    foo: String,
    card: Card,
    billing_details: BillingDetails,
}
```
`card` is a JS type and `BillingDetails` is a struct.

However, the following results in the following from the compiler:
`error: expected an inert attribute, found an attribute macro`

I have also tried:
```rust
#[wasm_bindgen]
pub struct PaymentMethodData{
    r#type: String,
    card: Arc<Card>,
    billing_details: BillingDetails,
}
```
Which does compile but stripe throws a runtime error (`IntegrationError: Invalid value for createPaymentMethod: type should be string. You specified: undefined.`).

Any ideas on where to go from here?

**Comment by alexcrichton** [comment]
I don't think that private fields are exported into JS, so if you're passing this to an API that expects to be able to access a `type` property then that won't work? You'll need to probably make an `impl` with methods that work as accessors?

**Comment by disconsented** [comment]
Okay I've finally figured out what is going on, internally stripe elements is accessing using bracket notation which is where the issue has cropped up from.

```javascript
var i = r[o]
```

Where `r` is the object and `o` is a string with a field name, is there a way to expose fields for this or would I need to write a small translation layer?

**Comment by alexcrichton** [comment]
Ah that makes sense! I'm unfortunately not enough of a JS-wizard to know what the options available to wasm-bindgen are to solve that. I'd probably have a small layer for now to translate.
